jujugui/static/gui/src/app/store/env/bakery.js

```diff
@@ -35,260 +35,267 @@ YUI.add('juju-env-bakery', function(Y) {
    * Bakery client inspired by the equivalent GO code.
    *
    * This object exposes the ability to perform requests
-   * that automatically acquire and discharge macaroons
+   * that automatically acquire and discharge macaroons.
    *
    * @class Bakery
    */
 
   var Bakery = Y.Base.create('Bakery',
-      Y.Base, [], {
-
-      /**
-       Initialize.
-
-       @method initializer
-       @param {WebHandler} The web handler used to send request.
-       @param {Function} The visit method that would be used for interaction.
-       @return {undefined} Nothing.
-       */
-      initializer: function (cfg) {
-        this.webhandler = cfg.webhandler;
-        if (cfg.visitMethod == null) {
-          this.visitMethod = this._defaultVisitMethod;
-        } else {
-          this.visitMethod = cfg.visitMethod;
-        }
-        this.macaroonName = 'Macaroons-' + cfg.serviceName;
-        this.setCookiePath = cfg.setCookiePath;
-        this.nonceLen = 24;
-      },
-
-      /**
-       Takes the path supplied by the caller and makes a get request to the
-       requestHandlerWithInteraction instance. If setCookiePath is set then
-       it is used to set a cookie back to the ui after authentication.
-
-       @param {String} The path to make the api request to.
-       @param {Function} successCallback Called when the api request completes
-       successfully.
-       @param {Function} failureCallback Called when the api request fails
-       with a response of >= 400 except 401/407 where it does authentication.
-       */
-      sendGetRequest: function (path, successCallback, failureCallback) {
+    Y.Base, [], {
+
+    /**
+     Initialize.
+
+     @method initializer
+     @param {Object} cfg Application configuration data.
```

Comment:
Please document what is expected to be in this. How is the caller meant to know what to pass in?